app-misc/mosquitto/mosquitto-1.4.15-r1.ebuild

```diff
@@ -0,0 +1,100 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+PYTHON_COMPAT=( python2_7 )
+
+inherit systemd user toolchain-funcs python-any-r1
+
+DESCRIPTION="An Open Source MQTT v3 Broker"
+HOMEPAGE="http://mosquitto.org/"
+SRC_URI="http://mosquitto.org/files/source/${P}.tar.gz"
+
+LICENSE="EPL-1.0"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~x86"
+IUSE="bridge examples +persistence +srv ssl tcpd test websockets"
+
+REQUIRED_USE="test? ( bridge )"
+
+RDEPEND="tcpd? ( sys-apps/tcp-wrappers )
+	ssl? ( dev-libs/openssl:0= )"
+DEPEND="${RDEPEND}
+	${PYTHON_DEPS}
+	srv? ( net-dns/c-ares )
+	websockets? ( net-libs/libwebsockets )"
+
+_emake() {
+	LIBDIR=$(get_libdir)
+	emake \
+		CC="$(tc-getCC)" \
+		LIB_SUFFIX="${LIBDIR:3}" \
+		WITH_BRIDGE="$(usex bridge)" \
+		WITH_PERSISTENCE="$(usex persistence)" \
+		WITH_SRV="$(usex srv)" \
+		WITH_TLS="$(usex ssl)" \
+		WITH_WEBSOCKETS="$(usex websockets)" \
+		WITH_WRAP="$(usex tcpd)" \
+		"$@"
+}
+
+pkg_setup() {
+	enewgroup mosquitto
+	enewuser mosquitto -1 -1 -1 mosquitto
+}
+
+src_prepare() {
+	eapply "${FILESDIR}/${P}-conditional-tests.patch"
```

Comment:
Looks like you haven't understand the problem from [bug 653238](https://bugs.gentoo.org/653238). Have you actually tried to run you your ebuild? This is the always the last thing you should do before creating a PR...

Reply:
Correct, I was confused by that bug. I did test my ebuild though. I will work on fixing this up today.

Reply:
The problem is that you named the patch ***mosquitto-1.4.15-r1**-conditional-tests.patch*. However, you referenced the patch via *$P* which will resolve to *mosquitto-1.4.15* (*$PVR* would resolve to *mosquitto-1.4.15-r1* however please don't use that!). So the solution is to rename patch to *mosquitto-1.4.15-conditional-tests.patch*.

And for future please make sure you run the ebuild before creating a PR, i.e. cd into the folder containing the ebuild and run `ebuild <ebuild file> clean install` to make sure the ebuild itself actually works and to catch such problems.

Reply:
Thanks for the info!